presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolAllocator.java

```diff
@@ -51,6 +52,11 @@ public Symbol newSymbol(Symbol symbolHint)
         return newSymbol(symbolHint.getName(), symbols.get(symbolHint));
     }
 
+    public Symbol newSymbol(QualifiedName nameHint, Type type)
+    {
+        return newSymbol(nameHint.getSuffix(), type, null);
```

Comment:
Why suffix here? (not prefix or other parts?)

Reply:
it's something like `catalog.schema.name` so it's the last part of the `QualifiedName`.

Reply:
ok cool. Maybe add a comment to explain.

Reply:
I don't think this needs a comment. It's based on the pretty much the definition of `QualifiedName`. There's nothing special or unexpected. In general I don't think we add comment unless there's something the code itself doesn't explain or doesn't make sense.

Reply:
I agree with RongRong here, since it is a nameHint, it is quite self explaining why we just use the last part. But to make it more clear, maybe rename it to nameHintForFunction (or not, should be fine for both)?